lib/rucio/tests/test_replica.py

```diff
@@ -257,15 +257,12 @@ def test_delete_replicas_from_datasets(self, rse_factory, mock_scope, root_accou
         _, rse_id = rse_factory.make_mock_rse()
```

Comment:
Seems like this test was explicitly ensuring that open datasets are not removed, while closed one are removed. Going all the way back to 2013 ... https://github.com/rucio/rucio/commit/c93322d0641f36f165288e1a9477bf83f8ca1692 . It's sad that there is absolutely no context in the commit message to understand why it was done like that.

Reply:
@mlassnig and @bari12 should comment. AFAIR, the purpose was to address this kind of workflow: You have a dataset, all the rules for this dataset expires, all the replica from the dataset are removed, then the dataset is deleted and the dataset name cannot be reused. If the dataset is open that means that the production system might try to add new files, so we don't want that the dataset completely disappear.
But I don't really think that it's useful to keep the dataset if all the other files are already gone.

Reply:
I think that depends, e.g., for user datasets it makes sense, for officially produced maybe not (as the associated task has finished/failed anyway).

Reply:
Yeah, but it leaves a lot of zombies, for maybe one or two users datasets. I would really prefer to delete everything. Let's discuss in tomorrow devops

Reply:
I agree, it probably was the use-case that an open dataset might be re-used. For user ones it actually makes sense.
The zombies I am not too worried about, at least in terms of DB rows used, since the total stays the same with the row moved to `deleted_dids`.